test/extensions/transport_sockets/tls/cert_validator/san_matcher_config_test.cc

```diff
@@ -0,0 +1,69 @@
+#include "envoy/extensions/transport_sockets/tls/v3/common.pb.validate.h"
+
+#include "source/common/protobuf/message_validator_impl.h"
+#include "source/common/protobuf/utility.h"
+#include "source/extensions/transport_sockets/tls/cert_validator/san_matcher_config.h"
+
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Tls {
+
+// Verify that we handle an invalid san type enum.
+TEST(SanMatcherConfigTest, TestInvalidSanType) {
+  envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher san_matcher;
+  san_matcher.mutable_matcher()->set_exact("foo.example");
+  san_matcher.set_san_type(
+      envoy::extensions::transport_sockets::tls::v3::
+          SubjectAltNameMatcher_SanType_SubjectAltNameMatcher_SanType_INT_MIN_SENTINEL_DO_NOT_USE_);
+  const SanMatcherPtr matcher = createStringSanMatcher(san_matcher);
```

Comment:
I added the test case because CI failed due to of coverage numbers reducing (as NOT_REACHED_GCOVR_EXCL_LINE does not work as advertised anymore).
I'll change it to an exception if that sounds okay.